File: shamelagpt-ios/shamelagpt/Core/Networking/Models/UserResponse.swift
//
//  UserResponse.swift
//  ShamelaGPT
//
//  Created by Codex on 05/12/2025.
//

import Foundation

struct UserResponse: Codable, Equatable {
    let id: String
    let firebaseUid: String
    let email: String?
    let displayName: String?
    let createdAt: String
    let updatedAt: String
    let lastLogin: String?
}

File: shamelagpt-android/app/src/main/java/com/shamelagpt/android/data/remote/dto/UserResponse.kt
package com.shamelagpt.android.data.remote.dto

import com.google.gson.annotations.SerializedName

/**
 * Response payload for user profile.
 */
data class UserResponse(
    val id: String,
    @SerializedName("firebase_uid")
    val firebaseUid: String,
    val email: String?,
    @SerializedName("display_name")
    val displayName: String?,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String,
    @SerializedName("last_login")
    val lastLogin: String?
)
